app/models/searchable.rb

```diff
@@ -129,6 +129,12 @@ def search_index
   end
 
   def can_index_in_search?
-    self.class.searchable_instances.find_by(id: self.id).present? && Whitehall.searchable_classes.include?(self.class)
+    is_searchable? && Whitehall.searchable_classes.include?(self.class)
+  end
+
+  # this could be avoided by merging the two Active Record Relations into another
+  # ActiveRecord Relation in model/edition.rb .search_only
```

Comment:
I haven't looked into the alternative but iterating over all searchable instances is going to be much slower than the previous version with the find_by. It's a bit surprising to have that happen in this method, since an instance shouldn't really need to know about anything else to work out if its indexable or not.

Minor nitpick: I'm not sure about extracting this method... "is_searchable?" and "can_index_in_search?" sound more or less the same to me. And it's complicated further by withdrawn items not actually being searchable by default to the end user, even though they are going to be part of the index.